datafusion/core/src/execution/context.rs

```diff
@@ -1118,10 +1117,24 @@ impl SessionConfig {
         self
     }
 
-    /// Convert configuration to name-value pairs
+    /// Get the currently configured batch size
+    pub fn batch_size(&self) -> usize {
+        self.config_options.get_u32(OPT_BATCH_SIZE) as usize
+    }
+
+    /// Get the current configuration options
+    pub fn config_options(&self) -> &ConfigOptions {
+        &self.config_options
+    }
+
+    /// Convert configuration options to name-value pairs with values converted to strings. Note
+    /// that this method will eventually be deprecated and replaced by [config_options].
     pub fn to_props(&self) -> HashMap<String, String> {
         let mut map = HashMap::new();
-        map.insert(BATCH_SIZE.to_owned(), format!("{}", self.batch_size));
+        map.insert(
+            BATCH_SIZE.to_owned(),
+            format!("{}", self.config_options.get_u32(OPT_BATCH_SIZE)),
+        );
```

Comment:
I wonder if it is possible to avoid having to enumerate all options again in this function -- what about maybe  converting all entries in `self.config_options`?

Reply:
Should we consolidate `BATCH_SIZE` with `OPT_BATCH_SIZE`?

Reply:
We should.  Ballista currently relies on this method and the keys used here. Once all the properties are converted I will update Ballista to use the new `config_option` method and delete this method,

Reply:
So we use these names internally as well so I ended up removing the duplicate option. This will require Ballista to look for the new name but that is a small change, I filed https://github.com/apache/arrow-ballista/issues/73 to track this.